src/app/core/components/widgets/widget-network/widget-network.component.ts

```diff
@@ -0,0 +1,349 @@
+        if (nicName in this.nicInfoMap) {
+          const sent = this.utils.convert(evt.data.sent_bytes_rate);
+          const received = this.utils.convert(evt.data.received_bytes_rate);
+
+          const nicInfo = this.nicInfoMap[nicName];
+          nicInfo.in = received.value + ' ' + received.units + '/s';
+          nicInfo.out = sent.value + ' ' + sent.units + '/s';
+
+          if (evt.data.sent_bytes - nicInfo.lastSent > 1024) {
+            nicInfo.lastSent = evt.data.sent_bytes;
+            this.tableService.updateStateInfoIcon(nicName, 'sent');
+          }
+
+          if (evt.data.received_bytes - nicInfo.lastReceived > 1024) {
+            nicInfo.lastReceived = evt.data.received_bytes;
+            this.tableService.updateStateInfoIcon(nicName, 'received');
+          }
+        }
+      }
+    });
+  }
+
+  getColspan(index: number): number {
+    let colSpan = 6;
+    if (this.availableNics.length <= 3) {
+      colSpan = 6;
+    } else if (this.availableNics.length == 4) {
+      colSpan = 3;
+    } else if (this.availableNics.length == 5) {
+      if (index < 2) {
+        colSpan = 3;
+      } else {
+        colSpan = 2;
+      }
+    } else if (this.availableNics.length >= 6) {
+      colSpan = 2;
+    }
+    return colSpan;
+  }
+
+  updateMapInfo(): void {
+    this.availableNics.forEach((nic: BaseNetworkInterface) => {
+      this.nicInfoMap[nic.state.name] = {
+        ip: this.getIpAddress(nic),
+        state: this.getLinkState(nic),
+        in: '',
+        out: '',
+        lastSent: 0,
+        lastReceived: 0,
+        chartData: null,
+      };
+    });
+  }
+
+  updateGridInfo(): void {
+    const nicsCount = this.availableNics.length;
+    let maxTicksLimit = 5;
+
+    if (nicsCount <= 3) {
+      this.rows = nicsCount;
+      if (nicsCount == 3) {
+        this.paddingTop = 0;
+        this.paddingBottom = 4;
+        this.gap = 8;
+        this.aspectRatio = 336 / 100;
+        maxTicksLimit = 3;
+      } else {
+        this.paddingTop = 16;
+        this.paddingBottom = 16;
+        this.gap = 16;
+
+        if (nicsCount == 2) {
+          this.aspectRatio = 304 / 124;
+          maxTicksLimit = 3;
+        } else {
+          this.aspectRatio = 474 / 200;
+        }
+      }
+    } else {
+      this.rows = 2;
+      this.paddingTop = 16;
+      this.paddingBottom = 16;
+      this.gap = 16;
+    }
+
+    if (this.rows < 1) {
+      this.rows++;
+    } else if (this.rows > 3) {
+      this.rows = 3;
+    }
+    const space = (this.rows - 1) * this.gap + this.paddingTop + this.paddingBottom;
+    this.rowHeight = (this.contentHeight - space) / this.rows;
+
+    const newChartOptions = { ...this.chartOptions };
+    newChartOptions.scales.yAxes[0].ticks.maxTicksLimit = maxTicksLimit;
+    newChartOptions.aspectRatio = this.aspectRatio;
+    this.chartOptions = newChartOptions;
+  }
+
+  getIpAddress(nic: BaseNetworkInterface): string {
+    let ip = T('Unknown');
```

Comment:
I was confused by "Unknown" string. Maybe we could specify that it belongs to IP Address?

Reply:
![image](https://user-images.githubusercontent.com/351613/125600660-a7d04499-5249-48c1-a519-dfac9b9b37d2.png)